src/Microsoft.Extensions.Options/InitializeOptions.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Extensions.Options
+{
+    /// <summary>
+    /// Implementation of IInitializeNamedOptions.
+    /// </summary>
+    /// <typeparam name="TOptions"></typeparam>
+    public class InitializeOptions<TOptions> : IInitializeOptions<TOptions> where TOptions : class
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="name">The name of the options.</param>
+        /// <param name="action">The action to register.</param>
+        public InitializeOptions(string name, Action<TOptions> action)
+        {
+            Name = name;
```

Comment:
null name signify's applying to all names.
string.empty is used to target the 'default' name, i.e. the overloads that don't specify name

Passing in a null action is equivalent to passing in a no-op o => { }, so I don't think its something we explicitly need to block (we didn't for the existing https://github.com/aspnet/Options/blob/dev/src/Microsoft.Extensions.Options/ConfigureOptions.cs, so we need to be consistent)  Note: we do require non null actions on the sugar `Configure/Initialize` apis since it doesn't make sense to add no-op/null actions there.